vassal-app/src/main/java/VASSAL/build/module/PrivateMap.java

```diff
@@ -116,10 +116,15 @@ else if (SIDE.equals(key)) {
       owners = (String[]) value;
     }
     else if (USE_BOARDS.equals(key)) {
-      for (final Map m : Map.getMapList()) {
-        if (m.getMapName().equals(value)) {
-          surrogate = m;
-          break;
+      if (value == null || ((String) value).isEmpty()) {
```

Comment:
Do you need to check that it's an instance of String here?

Reply:
True, unlikely, but possible to cause a problem. Update pushed